tritonsrc/attn_torch_function.py

```diff
@@ -924,109 +925,123 @@ def backward_fused(ctx, do, _, fwd_tuning_result):
-                    num_head_q=num_head_q,
-                    num_head_k=num_head_k,
-                    cu_seqlens_q=null_tensor,
-                    cu_seqlens_k=null_tensor,
-                    num_seqlens=0,
-                    max_seqlen_q=q.shape[2],
-                    max_seqlen_k=k.shape[2],
-                    head_dim=Lk,
-                    dropout_p=ctx.dropout_p,
-                    philox_seed_ptr=philox_seed,
-                    philox_offset1=philox_offset,
-                    philox_offset2=0,
-                    BLOCK_DMODEL=head_dim_rounded,
-                    CAUSAL=ctx.causal,
-                    ENABLE_DROPOUT=ctx.dropout_p > 0.0,
-                    PADDED_HEAD=padded_head,
-                    BIAS_TYPE=ctx.bias_type,
-                    )
-            report = attn_extra_args.report_best_config
-            if report:
-                best = copy.deepcopy(tuned_attn_bwd.best_config)
-                attn_extra_args.report_best_config('attn_bwd', best)
-        else:
-            bare_attn_bwd[grid](
-                    q, k, v, b, ctx.sm_scale,
-                    o, do,
-                    dk, dv, dq, db,
-                    L, delta,
-                    q.stride(0), q.stride(1), q.stride(2), q.stride(3),
-                    k.stride(0), k.stride(1), k.stride(2), k.stride(3),
-                    v.stride(0), v.stride(1), v.stride(2), v.stride(3),
-                    b.stride(0), b.stride(1), b.stride(2), b.stride(3),
-                    do.stride(0), do.stride(1), do.stride(2), do.stride(3),
-                    dk.stride(0), dk.stride(1), dk.stride(2), dk.stride(3),
-                    dv.stride(0), dv.stride(1), dv.stride(2), dv.stride(3),
-                    dq.stride(0), dq.stride(1), dq.stride(2), dq.stride(3),
-                    stride_dbz, stride_dbh, stride_dbm, stride_dbn,  # db may be empty
-                    num_head_q=num_head_q,
-                    num_head_k=num_head_k,
-                    cu_seqlens_q=null_tensor,
-                    cu_seqlens_k=null_tensor,
-                    num_seqlens=0,
-                    max_seqlen_q=q.shape[2],
-                    max_seqlen_k=k.shape[2],
-                    head_dim=Lk,
-                    dropout_p=ctx.dropout_p,
-                    philox_seed_ptr=philox_seed,
-                    philox_offset1=philox_offset,
-                    philox_offset2=0,
-                    BLOCK_DMODEL=head_dim_rounded,
-                    CAUSAL=ctx.causal,
-                    ENABLE_DROPOUT=ctx.dropout_p > 0.0,
-                    PADDED_HEAD=padded_head,
-                    BIAS_TYPE=ctx.bias_type,
-                    BLOCK_M1=BLOCK_M1,
-                    BLOCK_N1=BLOCK_N1,
-                    BLOCK_M2=BLOCK_M2,
-                    BLOCK_N2=BLOCK_N2,
-                    BLK_SLICE_FACTOR=BLK_SLICE_FACTOR,
-                    )
+        if k.requires_grad and v.requires_grad and q.requires_grad:
+            if ctx.autotune:
```

Comment:
No need to add Triton autotune to split kernel. The autotune function in `tritonsrc` is broken for quite some time.

Just leave an `assert False` here and remove actual kernel invocations